src/main/java/org/broadinstitute/hellbender/utils/test/DataprocTestUtils.java

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.hellbender.utils.test;
+
+import org.broadinstitute.hellbender.utils.runtime.ProcessController;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * utilities to help run tests on a Dataproc cluster
+ */
+public final class DataprocTestUtils {
+
+    private DataprocTestUtils(){}
+
+    /**
+     * @return the value of the GCLOUD environment variable if present, or "gcloud"
+     */
+    public static String getGCloudPath(){
+        final String gcloudBinDir = System.getenv("GCLOUD_HOME");
+        return gcloudBinDir == null ? "gcloud" : gcloudBinDir + "/gcloud";
+    }
+
+    /**
+     * create a minimal dataproc cluster which will shutdown automatically after 10 minutes of idling or 30 minutes
+     * @return the name of the cluster
+     */
+    public static String createTestCluster() {
+        final String clusterName = "gatk-test-" + UUID.randomUUID();
+        final String[] command = new String[]{
+                getGCloudPath(), "beta", "dataproc", "clusters", "create",
+                "--max-idle", "10m",
+                "--max-age", "30m",
+                "--num-workers", "2",
+                "--master-machine-type", "n1-highmem-2",
+                "--worker-machine-type", "n1-highmem-2",
+                clusterName
+        };
+        BaseTest.runProcess(ProcessController.getThreadLocal(), command, "Couldn't create dataproc cluster");
+        return clusterName;
+    }
+
+    /**
+     * run a gatk spark tool on a dataproc cluster and throw if it fails
+     * @param tool name of the tool
+     * @param args arguments to the tool
+     * @param clusterName name of the cluster
+     */
+    public static void launchGatkTool(final String tool, final List<String> args, final String clusterName){
+            final List<String> command = new ArrayList<>();
+            command.add("./gatk-launch");
```

Comment:
This is a repo-dependent file - this method should live in the src/test to avoid again problems with the test framework in donwstream projects. I recommend to move this method to `GATKBaseTest` instead, to be sure that it is only run within GATK-tests...

Reply:
Good point.  I don't want to separate this method from the cluster creation method though, since they're sort of tied together in how you use them and I think having them together makes sense.  I'm happy to move the whole class to test where it wouldn't be packaged at all, or to document it clearly saying it won't work for downstream projects.  The whole command is extremely brittle and dependent on the exact launcher script you're using, so parameterizing it wouldn't help.  

Which would you prefer?

Reply:
I would go for packaging in src/test...